src/Microsoft.AspNetCore.DataProtection.Abstractions/project.json

```diff
@@ -14,6 +14,8 @@
       "type": "build",
       "version": "1.0.0-*"
     },
+    "Microsoft.AspNetCore.Hosting.Abstractions": "1.0.0-*",
```

Comment:
@smbecker had the same concern. I could move the method into the `DataProtection` to fix this. Alternatively, I could clean up how `GetApplicationUniqueIdentifier` works so that it doesn't have to know of `IApplicationEnvironment` \ `IHostingEnvironment`. I was about to file a RTM bug to clean up that bit.